apps/api/src/app/db/ratings/ratings.store.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { RatingsDocument } from './ratings.schema';
+import { JwtPayload } from '@dragonfish/shared/models/auth';
+import { isNullOrUndefined } from '@dragonfish/shared/functions';
+import { RatingOption } from '@dragonfish/shared/models/reading-history';
+import { ContentStore } from '../content';
+
+@Injectable()
+export class RatingsStore {
+    constructor(
+        @InjectModel('Ratings') private readonly ratings: Model<RatingsDocument>,
+        private readonly contentStore: ContentStore,
+    ) {}
+
+    public async addOrFetchRatingsDoc(user: JwtPayload, contentId: string) {
+        const existingDoc = await this.ratings.findOne({ contentId: contentId, userId: user.sub });
+
+        if (isNullOrUndefined(existingDoc)) {
+            const newDoc = new this.ratings({
+                contentId: contentId,
+                userId: user.sub,
+            });
+
+            return newDoc.save();
+        } else {
+            return existingDoc;
+        }
+    }
+
+    public async addLike(user: JwtPayload, contentId: string) {
+        const doc = await this.ratings.findOne({ contentId: contentId, userId: user.sub });
+        doc.rating = RatingOption.Liked;
+
+        return doc.save().then(async (doc) => {
+            await this.updateCounts(contentId);
+            return doc;
+        });
+    }
+
+    public async addDislike(user: JwtPayload, contentId: string) {
+        const doc = await this.ratings.findOne({ contentId: contentId, userId: user.sub });
+        doc.rating = RatingOption.Disliked;
+
+        return doc.save().then(async (doc) => {
+            await this.updateCounts(contentId);
+            return doc;
+        });
+    }
+
+    public async setNoVote(user: JwtPayload, contentId: string) {
+        const doc = await this.ratings.findOne({ contentId: contentId, userId: user.sub });
+        doc.rating = RatingOption.NoVote;
+
+        return doc.save().then(async (doc) => {
+            await this.updateCounts(contentId);
+            return doc;
+        });
+    }
+
+    private async updateCounts(contentId: string) {
+        const newLikedCount = await this.ratings
+            .countDocuments({ contentId: contentId, rating: RatingOption.Liked });
+        const newDislikedCount = await this.ratings
+            .countDocuments({ contentId: contentId, rating: RatingOption.Disliked });
+        console.log(`New Liked Count: ${newLikedCount}\nNew Disliked Count: ${newDislikedCount}`);
+        await this.contentStore.updateRatingCounts(contentId, newLikedCount, newDislikedCount);
```

Comment:
Nnnngaahh. This really makes me nervous. It couples the implementation of the `contentStore` and this `ratingStore` really tightly together.

I think that, instead, there should be a `ratingService` in front of this that's responsible for 1) calling `addLike`/`addDislike`/`setNoVote` in this Store, and 2) updating the rating counts over in `Content`, and then the this store could _JUST_ update and return its own documents.